subsys/net/openthread/rpc/client/ot_rpc_if_client.c

```diff
@@ -166,11 +168,26 @@ static int ot_rpc_l2_enable(struct net_if *iface, bool state)
 {
 	const size_t cbor_buffer_size = 1;
 	struct nrf_rpc_cbor_ctx ctx;
+	struct zcbor_string zst;
+	int error = 0;
 
 	if (!state) {
 		otRemoveStateChangeCallback(NULL, ot_state_changed_handler, iface);
 	}
 
+	net_if_set_link_addr(iface, mac, 8, NET_LINK_IEEE802154);
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, 0);
+	nrf_rpc_cbor_cmd_rsp(&ot_group, OT_RPC_CMD_IF_EXTADDR, &ctx);
+
+	if (!zcbor_bstr_decode(ctx.zs, &zst)) {
+		error = -1;
```

Comment:
```suggestion
		error = -EINVAL;
```

Reply:
Also, you have always call `nrf_rpc_cbor_decoding_done` or the app will hang otherwise.
Checking the length might be a good idea as well.

Reply:
fixed.